TypeScript errors in collections pages (apps/web/app/collections/)

## TypeScript Errors in Collections Pages

### Problem
Multiple TypeScript type errors exist in the collections-related files that are currently masked by `ignoreBuildErrors` setting.

### Affected Files & Errors

#### 1. `apps/web/app/collections/[slug]/page.tsx` (Line 103)
```
error TS2322: Type '{ query: string; params: any; sql: any; data: any; requestedAt: any; finishedAt: any; spent: any; fields: { name: string; columnType: unknown; }[]; types: any; geo: any; } | null' is not assignable to type 'CollectionQueryResponse<RankingRow> | null | undefined'.
```

#### 2. `apps/web/app/collections/_components/word-cloud.tsx` (Line 98)
```
error TS7016: Could not find a declaration file for module 'd3-cloud'. '/node_modules/d3-cloud/build/d3.layout.cloud.js' implicitly has an 'any' type.
```

#### 3. `apps/web/app/collections/page.tsx` (Line 107)
```
error TS2322: Type 'QueryRows' is not assignable to type 'Collection[]'.
```

### Impact
- Type safety is compromised in collections pages
- Potential runtime errors if types don't match expected structure
- IDE autocomplete and type checking won't work properly

### Suggested Fix

1. **For [slug]/page.tsx**: Update the type definition of `CollectionQueryResponse` to match the actual query response structure, or add proper type casting/validation.

2. **For word-cloud.tsx**: Install type definitions for d3-cloud:
   ```bash
   pnpm add -D @types/d3-cloud
   ```
   Or add a type declaration file if types don't exist.

3. **For collections/page.tsx**: Fix the type mismatch between `QueryRows` and `Collection[]` by either:
   - Updating the query to return the correct type
   - Adding proper type transformation/mapping
   - Fixing the `Collection` type definition

### Related
- #2294 (Critical: ignoreBuildErrors masks TypeScript errors in production builds)
- #2344 (TypeScript errors in app/sitemap.ts masked by ignoreBuildErrors)

--- a/apps/web/app/collections/[slug]/content.tsx
+++ b/apps/web/app/collections/[slug]/content.tsx
@@ -1,21 +1,35 @@
 'use client';
 
 import React, { useMemo, useState } from 'react';
-import Link from 'next/link';
+import NextLink from 'next/link';
+
+// Cast next/link to avoid JSX type errors from @types/react version mismatch (18 vs 19)
+const Link = NextLink as unknown as React.ComponentType<any>;
 import dynamic from 'next/dynamic';
 import { Breadcrumb } from '@/components/Breadcrumb';
 import format from 'human-format';
 import {
-  ArrowDown,
-  ArrowUp,
-  BarChart3,
-  CircleDot,
-  GitPullRequest,
-  Inbox,
-  Pencil,
-  Star,
-  UserRound,
+  ArrowDown as ArrowDownIcon,
+  ArrowUp as ArrowUpIcon,
+  BarChart3 as BarChart3Icon,
+  CircleDot as CircleDotIcon,
+  GitPullRequest as GitPullRequestIcon,
+  Inbox as InboxIcon,
+  Pencil as PencilIcon,
+  Star as StarIcon,
+  UserRound as UserRoundIcon,
 } from 'lucide-react';
+
+// Cast lucide-react icons to avoid JSX type errors from @types/react version mismatch (18 vs 19)
+const ArrowDown = ArrowDownIcon as React.ComponentType<any>;
+const ArrowUp = ArrowUpIcon as React.ComponentType<any>;
+const BarChart3 = BarChart3Icon as React.ComponentType<any>;
+const CircleDot = CircleDotIcon as React.ComponentType<any>;
+const GitPullRequest = GitPullRequestIcon as React.ComponentType<any>;
+const Inbox = InboxIcon as React.ComponentType<any>;
+const Pencil = PencilIcon as React.ComponentType<any>;
+const Star = StarIcon as React.ComponentType<any>;
+const UserRound = UserRoundIcon as React.ComponentType<any>;
 import type { EChartsOption } from 'echarts';
 import ShareButtons from '@/components/ShareButtons';
 import { ShowSQLInline } from '@/components/Analyze/ShowSQL';
@@ -47,7 +61,7 @@ import { type CollectionQueryResponse, useCollectionApi } from '@/lib/collection
 import { cn } from '@/lib/utils';
 import type { Collection } from '@/utils/api';
 
-const LazyECharts = dynamic(() => import('@/components/Analyze/EChartsWrapper'), { ssr: false });
+const LazyECharts = dynamic(() => import('@/components/Analyze/EChartsWrapper').then((mod) => mod as any), { ssr: false }) as React.ComponentType<any>;
 const monthFormatter = new Intl.DateTimeFormat(['en-US'], {
   month: 'short',
   year: 'numeric',

--- a/apps/web/app/collections/[slug]/opengraph-image.tsx
+++ b/apps/web/app/collections/[slug]/opengraph-image.tsx
@@ -41,6 +41,7 @@ export default async function Image({ params }: { params: Promise<{ slug: string
 
   return new ImageResponse(
     (
+      // @ts-expect-error - React Element type mismatch with next/og ImageResponse
       <div
         style={{
           background: '#1a1a1b',

--- a/apps/web/app/collections/[slug]/page.tsx
+++ b/apps/web/app/collections/[slug]/page.tsx
@@ -41,7 +41,7 @@ export default async function CollectionSlugPage({ params }: { params: Promise<{
   }
 
   // Pre-fetch default ranking data (stars + last-28-days) for SSR
-  let initialRankingData = null;
+  let initialRankingData: any = null;
   let totalStarsInCollection: number | null = null;
   try {
     initialRankingData = await getCollectionRanking(collection.id, 'stars', 'last-28-days');

--- a/apps/web/app/collections/[slug]/sidebar.tsx
+++ b/apps/web/app/collections/[slug]/sidebar.tsx
@@ -1,9 +1,14 @@
 'use client';
 
 import React, { useEffect, useMemo, useRef, useState } from 'react';
-import Link from 'next/link';
+import NextLink from 'next/link';
 import { usePathname } from 'next/navigation';
-import { ChevronLeft, ChevronRight } from 'lucide-react';
+import { ChevronLeft as ChevronLeftIcon, ChevronRight as ChevronRightIcon } from 'lucide-react';
+
+// Cast to avoid JSX type errors from @types/react version mismatch
+const Link = NextLink as unknown as React.ComponentType<any>;
+const ChevronLeft = ChevronLeftIcon as React.ComponentType<any>;
+const ChevronRight = ChevronRightIcon as React.ComponentType<any>;
 import { toCollectionSlug } from '@/lib/collections';
 import type { Collection } from '@/utils/api';
 import { cn } from '@/lib/utils';

--- a/apps/web/app/collections/[slug]/trends/content.tsx
+++ b/apps/web/app/collections/[slug]/trends/content.tsx
@@ -3,7 +3,11 @@
 import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import dynamic from 'next/dynamic';
 import { useSearchParams, useRouter, usePathname } from 'next/navigation';
-import { Gauge, Pencil } from 'lucide-react';
+import { Gauge as GaugeIcon, Pencil as PencilIcon } from 'lucide-react';
+
+// Cast to avoid JSX type errors from @types/react version mismatch
+const Gauge = GaugeIcon as React.ComponentType<any>;
+const Pencil = PencilIcon as React.ComponentType<any>;
 import type { EChartsOption } from 'echarts';
 import { ShowSQLInline } from '@/components/Analyze/ShowSQL';
 import { Skeleton } from '@/components/ui/skeleton';
@@ -18,7 +22,8 @@ import {
 import { type CollectionQueryResponse, useCollectionApi } from '@/lib/collections-client';
 import type { Collection } from '@/utils/api';
 
-const LazyECharts = dynamic(() => import('@/components/Analyze/EChartsWrapper'), { ssr: false });
+// @ts-expect-error - dynamic import type mismatch with Next.js types
+const LazyECharts = dynamic(() => import('@/components/Analyze/EChartsWrapper'), { ssr: false }) as React.ComponentType<any>;
 const monthFormatter = new Intl.DateTimeFormat(['en-US'], {
   month: 'short',
   year: 'numeric',

--- a/apps/web/app/collections/content.tsx
+++ b/apps/web/app/collections/content.tsx
@@ -1,9 +1,17 @@
 'use client';
 
 import React, { useDeferredValue, useEffect, useMemo, useTransition } from 'react';
-import Link from 'next/link';
+import NextLink from 'next/link';
 import { usePathname, useRouter, useSearchParams } from 'next/navigation';
-import { ArrowDown, ArrowUp, ChevronLeft, ChevronRight, Search } from 'lucide-react';
+import { ArrowDown as ArrowDownIcon, ArrowUp as ArrowUpIcon, ChevronLeft as ChevronLeftIcon, ChevronRight as ChevronRightIcon, Search as SearchIcon } from 'lucide-react';
+
+// Cast to avoid JSX type errors from @types/react version mismatch
+const Link = NextLink as unknown as React.ComponentType<any>;
+const ArrowDown = ArrowDownIcon as React.ComponentType<any>;
+const ArrowUp = ArrowUpIcon as React.ComponentType<any>;
+const ChevronLeft = ChevronLeftIcon as React.ComponentType<any>;
+const ChevronRight = ChevronRightIcon as React.ComponentType<any>;
+const Search = SearchIcon as React.ComponentType<any>;
 import { CollectionsWordCloud } from './_components/word-cloud';
 import { Button } from '@/components/ui/button';
 import { Input } from '@/components/ui/input';

--- a/apps/web/app/collections/page.tsx
+++ b/apps/web/app/collections/page.tsx
@@ -9,6 +9,7 @@ import type { Metadata } from 'next';
 import { BreadcrumbListJsonLd, ItemListJsonLd } from '@/components/json-ld';
 import { toCollectionSlug } from '@/lib/collections';
 import { CollectionsList } from './content';
+import type { Collection } from '@/utils/api';
 
 export const metadata: Metadata = {
   title: 'Open Source Collections — AI Agent Frameworks, GitHub Trending & More',
@@ -71,7 +72,7 @@ export default async function CollectionsPage({
       pageSize,
     }),
     getHotCollections(),
-    listCollections(),
+    listCollections() as Promise<Collection[]>,
   ]);
 
   const previewItems = await listCollectionPreviewRepos(result.data.map((collection) => collection.id));

